modules/rostests/apitests/user32/ScrollBarRedraw.c

```diff
@@ -0,0 +1,722 @@
+                skip("Failed to retrieve client area dimensions, code: %ld\n", GetLastError());
+                return -1;
+            }
+            ClientWidth = Rect.right;
+            ClientHeight = Rect.bottom;
+
+            if (!GetWindowRect(Window, &Rect))
+            {
+                skip("Failed to retrieve window dimensions, code: %ld\n", GetLastError());
+                return -1;
+            }
+            OrigWidth  = Rect.right - Rect.left;
+            OrigHeight = Rect.bottom - Rect.top;
+
+            SmallWidth  = max((OrigWidth  * 3) / 4, 1);
+            SmallHeight = max((OrigHeight * 3) / 4, 1);
+            OrigWidth   = max(OrigWidth,  SmallWidth  + 1);
+            OrigHeight  = max(OrigHeight, SmallHeight + 1);
+
+            trace("OrigWidth: %d, OrigHeight: %d, SmallWidth: %d, SmallHeight: %d\n",
+                  OrigWidth, OrigHeight, SmallWidth, SmallHeight);
+
+            HideVertScrollBar(Window);
+            HideHorzScrollBar(Window);
+
+            FsmState = FSM_STATE_START;
+            FsmTimer = 0;
+
+            return 0;
+        }
+
+        case WM_PAINT:
+            if (FsmTimer == 0)
+            {
+                FsmTimer = SetTimer(Window, 1, FSM_STEP_PERIOD_MS, NULL);
+
+                if (FsmTimer == 0)
+                {
+                    skip("Failed to initialize FSM timer, code: %ld\n", GetLastError());
+                    DestroyWindow(Window);
+                    return 0;
+                }
+            }
+            return OnPaint(Window);
+
+        case WM_SIZE:
+        {
+            int NewWidth = LOWORD(lParam);
+            int NewHeight = HIWORD(lParam);
+
+            if (NewWidth != 0 && NewHeight != 0 &&
+                (NewWidth != ClientWidth || NewHeight != ClientHeight))
+            {
+                CurrentColor = (CurrentColor + 1) % TEST_COLOR_COUNT;
+                SetClassLongPtrW(Window,
+                                 GCLP_HBRBACKGROUND,
+                                 (LONG_PTR)ColorBrushes[CurrentColor]);
+
+                trace("New window size: %d x %d, new color: 0x%.8lX\n",
+                      NewWidth, NewHeight, Colors[CurrentColor]);
+
+                ClientWidth = NewWidth;
+                ClientHeight = NewHeight;
+            }
+            return 0;
+        }
+
+        case WM_ERASEBKGND:
+            /* We use WM_PAINT instead, since WM_ERASEBKGND is issued before WM_SIZE. */
+            return 1;
+
+        case WM_TIMER:
+            if (wParam != 0 && wParam == FsmTimer)
+            {
+                return FsmStep(Window);
+            }
+            break;
+
+        case WM_NCDESTROY:
+            if (FsmTimer != 0)
+            {
+                KillTimer(Window, FsmTimer);
+                FsmTimer = 0;
+
+                if (FsmState != FSM_STATE_END)
+                {
+                    skip("Window was closed before test concluded, FsmState: %d, FSM_STATE_END: %d.\n", FsmState, FSM_STATE_END);
```

Comment:
```suggestion
                    skip("Window closed before test concluded, FsmState: %d, FSM_STATE_END: %d\n", FsmState, FSM_STATE_END);
```
for example / as a start

Reply:
Yup yup, I've taken this suggestion.  I've also split up the formatted arguments onto the second line to fit the text into 100 columns.